### PR TITLE
Add `#![feature(const_fn_floating_point_arithmetic)]`

### PR DESCRIPTION
**Pull request by ecstatic-morse**
cc #76618 

This is a template for splitting up `const_fn` into granular feature gates. I think this will make it easier, both for us and for users, to track stabilization of each individual feature. We don't *have* to do this, however. We could also keep stabilizing things out from under `const_fn`.

cc @rust-lang/wg-const-eval 
r? @oli-obk

### CODE REVIEW

**Comment by ecstatic-morse** [comment]
This is a draft until I open a specific tracking issue for `const_fn_floating_point_arithmetic`. Currently it uses the meta `const_fn` one (#57563).

File: compiler/rustc_middle/src/ty/sty.rs [hunk 1]
```diff
@@ -2092,6 +2092,11 @@ impl<'tcx> TyS<'tcx> {
         }
     }
 
+    #[inline]
+    pub fn is_ptr(&self) -> bool {
```
Comment (RalfJung) [inline compiler/rustc_middle/src/ty/sty.rs:2096]: This is a confusing name, given that methods like `is_mutable_ptr` also include reference types. I do not have a good idea though, beyond `is_raw_or_fn_ptr`...

Also please add the new method next to the existing `is_*_ptr` methods to keep them all together.

**Comment by oli-obk** [comment]
> We don't have to do this, however. We could also keep stabilizing things out from under const_fn.

I'm in full agreement that we should do this and eliminate the `const_fn` feature gate entirely

r=me with Ralf's nit

File: compiler/rustc_middle/src/ty/sty.rs (hunk 1 above)
Comment (ecstatic-morse) [reply compiler/rustc_middle/src/ty/sty.rs]: Good point. A search for `is_ptr` only turned up `is_ptr_sized_integral` below. I've just removed this and used `is_unsafe_ptr || is_fnt_ptr`.

Comment (RalfJung) [reply compiler/rustc_middle/src/ty/sty.rs]: Yeah it is probably not worth having a helper for a single call site.

**Comment by ecstatic-morse** [comment]
cc #57241

**Pull request merged**